zowe/zowe-cli
# System tests should cleanup after themselves

**What It Does**
This PR enhances the test framework to simplify and automate the cleanup of test resources, such as datasets, USS files, jobs, and local files. The `TestEnvironment`, `ITestEnvironment`, and `TestUtils` classes have been updated to track and delete resources after system tests, preventing any leftover artifacts on the mainframe or local filesystem.

**How to Test : Run System Tests**
   - Execute the system tests as usual. Verify that all created resources (datasets, USS files, jobs, local files) are automatically cleaned up after the tests complete.
   - Ensure that no traces of jobs, USS files, or datasets are left behind on the mainframe.

**Review Checklist**
I certify that I have:

**Additional Comments**
- **Test Stability**
  - Occasionally, environment-specific issues may arise, especially with test datasets like IEFBR14 files. I'm not sure how to rectify this and definitely need areas using these files double-checked. 
- **Unchanged Files**: 
  - `zowe-cli\packages\zosjobs\__tests__\__system__\MonitorJobs.system.test.ts` only had minor indentation changes and was left intact as it leaves no jobs behind on the mainframe.
  - If a test file cleans up all resources, it was left without changes to avoid unnecessary modifications since most of these test files are large and would require many lines of renovations.
